### PR TITLE
refactor: refactor owner in smart contract account to connected signer

### PR DESCRIPTION
**Pull request by denniswon**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR updates the codebase to replace occurrences of `owner` with `signer` in various files for consistency and clarity.

### Detailed summary
- Replaced `owner` with `signer` in multiple files for consistency and clarity.
- Updated function calls, object properties, and variable names.
- Ensured alignment with the terminology used across the codebase.

> The following files were skipped due to too many changes: `packages/accounts/src/nani-account/transferNaniAccountOwnership.ts`, `packages/core/src/signer/local-account.ts`, `site/signers/eoa.md`, `packages/accounts/src/nani-account/__tests__/account.test.ts`, `site/packages/aa-core/smart-account-client/types/userOperationFeeOptions.md`, `site/snippets/aa-ethers/ethers-signer.ts`, `site/snippets/aa-ethers/ethers-provider.ts`, `site/packages/aa-accounts/light-account/actions/transferOwnership.md`, `site/using-smart-accounts/transfer-ownership/modular-account.md`, `site/signers/guides/walletkit.md`, `site/snippets/nani-account.ts`, `site/snippets/getting-started/send-user-operation.ts`, `site/snippets/aa-alchemy/light-account.ts`, `site/packages/aa-ethers/account-signer/signMessage.md`, `site/packages/aa-signers/magic/introduction.md`, `site/packages/aa-signers/portal/introduction.md`, `site/packages/aa-signers/capsule/introduction.md`, `site/packages/aa-signers/turnkey/introduction.md`, `site/packages/aa-signers/particle/introduction.md`, `site/packages/aa-signers/web3auth/introduction.md`, `site/packages/aa-ethers/provider-adapter/connectToAccount.md`, `site/packages/aa-signers/arcana-auth/introduction.md`, `packages/accounts/src/msca/plugins/multi-owner/signer.ts`, `site/packages/aa-signers/fireblocks/introduction.md`, `site/using-smart-accounts/session-keys/getting-started.md`, `site/snippets/getting-started/client.ts`, `site/getting-started/setup.md`, `packages/alchemy/src/client/lightAccountClient.ts`, `site/smart-accounts/index.md`, `site/snippets/session-keys/full-example.ts`, `site/signers/guides/custom-signer.md`, `site/signers/choosing-a-signer.md`, `packages/core/src/account/schema.ts`, `packages/ethers/src/__tests__/provider-adapter.test.ts`, `packages/accounts/src/light-account/decorator.ts`, `packages/accounts/src/light-account/actions/transferOwnership.ts`, `packages/accounts/src/light-account/client.ts`, `site/packages/aa-ethers/account-signer/connect.md`, `packages/alchemy/src/client/modularAccountClient.ts`, `site/smart-accounts/light-account/index.md`, `packages/core/src/account/types.ts`, `packages/ethers/e2e-tests/simple-account.e2e.test.ts`, `packages/core/src/account/base.ts`, `packages/alchemy/src/client/lightAccountClient.test.ts`, `packages/core/src/account/__tests__/simple.test.ts`, `packages/accounts/src/light-account/account.test.ts`, `site/using-smart-accounts/transfer-ownership/light-account.md`, `packages/accounts/src/msca/utils.ts`, `packages/accounts/src/msca/client.ts`, `site/packages/aa-ethers/account-signer/introduction.md`, `site/smart-accounts/modular-account/getting-started.md`, `packages/accounts/src/msca/account/multiOwnerAccount.ts`, `packages/alchemy/e2e-tests/light-account.e2e.test.ts`, `packages/core/src/account/simple.ts`, `site/resources/faqs.md`, `packages/accounts/src/light-account/account.ts`, `site/packages/aa-accounts/light-account/index.md`, `packages/accounts/src/nani-account/account.ts`, `packages/core/e2e-tests/simple-account.e2e.test.ts`, `packages/accounts/src/light-account/e2e-tests/light-account.e2e.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/accounts/src/light-account/account.ts [hunk 1]
```diff
@@ -32,24 +32,24 @@ import {
...
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+> = SmartContractAccountWithSigner<"LightAccount", TSigner> & {
   getLightAccountVersion: () => Promise<LightAccountVersion>;
   encodeTransferOwnership: (newOwner: Address) => Hex;
   getOwnerAddress: () => Promise<Address>;
-  setOwner: <TOwner extends SmartAccountSigner = SmartAccountSigner>(
-    newOwner: TOwner
+  setSigner: <TSigner extends SmartAccountSigner = SmartAccountSigner>(
+    signer: TSigner
   ) => void;
```
Comment (moldy530) [inline packages/accounts/src/light-account/account.ts:42]: While we're doing this refactor, can we remove the `setOwner` method? I kept it as mutable to match old behaviour, but mutability in this new context can result in some unexpected behaviour. it also makes building new features more difficult especially if you're using the signer of the account at all

File: packages/core/src/account/smartContractAccount.ts [hunk 2]
```diff
@@ -53,12 +53,12 @@ export type UpgradeToAndCallParams = {
...
-export type OwnedSmartContractAccount<
+export type SmartContractAccountWithSigner<
   Name extends string = string,
-  TOwner extends SmartAccountSigner = SmartAccountSigner
+  TSigner extends SmartAccountSigner = SmartAccountSigner
 > = SmartContractAccount<Name> & {
-  getOwner: () => TOwner;
-  setOwner: (owner: TOwner) => void;
+  getSigner: () => TSigner;
+  setSigner: (signer: TSigner) => void;
```
Comment (moldy530) [inline packages/core/src/account/smartContractAccount.ts:61]: from comment above:
```suggestion
```

File: site/packages/aa-ethers/provider-adapter/connectToAccount.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ head:
 
 # connectToAccount
 
-`connectToAccount` is a method on `EthersProviderAdapter` that you can optionally call to connect the provider to an account and returns a `AccountSigner`. This enables the returned `AccountSigner` to leverage the provider when signing messages, UserOperations, and transactions for a smart account using the owner account.
+`connectToAccount` is a method on `EthersProviderAdapter` that you can optionally call to connect the provider to an account and returns a `AccountSigner`. This enables the returned `AccountSigner` to leverage the provider when signing messages, UserOperations, and transactions for a smart account using the connected EOA signer account.
```
Comment (moldy530) [inline site/packages/aa-ethers/provider-adapter/connectToAccount.md:17]: a signer isn't required to be an EOA. just most of today's signers are, but a signer could also be a passkey signer for example

File: site/packages/aa-signers/arcana-auth/introduction.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ head:
 `ArcanaAuthSigner` provides implementations for all methods on `SmartAccountAuthenticator`:
 
 1.  [`authenticate`](/packages/aa-signers/arcana-auth/authenticate) -- supports user authentication.
-2.  [`getAddress`](/packages/aa-signers/arcana-auth/getAddress) -- supports typed data signatures from the smart contract account's owner address.
+2.  [`getAddress`](/packages/aa-signers/arcana-auth/getAddress) -- gets the address of the the smart contract account's connected EOA signer account.
```
Comment (moldy530) [inline site/packages/aa-signers/arcana-auth/introduction.md:22]: can we make sure to remove the EOA bit from other places? 

```suggestion
2.  [`getAddress`](/packages/aa-signers/arcana-auth/getAddress) -- gets the address of the the smart contract account's connected signer.
```

File: site/signers/choosing-a-signer.md [hunk 5]
```diff
@@ -24,7 +24,7 @@ A **Signer** is a service (e.g. Magic or Turnkey) or application (e.g. MetaMask)
 
 With Account Kit, you will deploy a **smart account** for each user instead of an EOA wallet. This smart account stores the user’s assets (e.g. tokens or NFTs). The default smart account in Account Kit is called [`LightAccount`](/smart-accounts/light-account/) and it uses a typical single-owner architecture.
 
-The smart account is controlled by an **Owner** address. The smart account will only execute a transaction if it was signed by the owner’s private key.
+The signer connected to the `SmartAccountClient` is used to sign messages, data including user operations and transactions. The signatures for the user operation will be only valid and execute if the signer is the owner or one of the owners of the account.
```
Comment (moldy530) [inline site/signers/choosing-a-signer.md:27]: signers aren't connected to a `SmartAccountClient`

Comment (moldy530) [reply site/signers/choosing-a-signer.md:27]: I think the previous description still holds here actually. maybe the only difference is that:
`The smart account is controlled by an Owner or potentially multiple Owner addresses` 

In code it makes sense to have a signer, but that signer represents an Owner. Even our MA has a default MultiOwner plugin installed, so the Owner concept is still relevant

File: packages/accounts/src/light-account/account.ts (hunk 1 above)
Comment (denniswon) [reply packages/accounts/src/light-account/account.ts:42]: this method is actually needed for typescript. yeah I know, was very weird, but get "[Type SmartAccountSigner  is assignable to the constraint of type 'TSigner', but 'TSigner' could be instantiated with a different subtype of constraint 'SmartAccountSigner']"
 
 Bascially, LightAccount needs to use the generic type TSigner in its interface in order for ts to not error.

Comment (denniswon) [reply packages/accounts/src/light-account/account.ts:42]: ah misunderstood the comment. okay will remove setSigner but due to above, might need to override getSigner instead.

File: site/packages/aa-ethers/provider-adapter/connectToAccount.md (hunk 3 above)
Comment (denniswon) [reply site/packages/aa-ethers/provider-adapter/connectToAccount.md:17]: good point.

File: packages/accounts/src/light-account/account.ts (hunk 1 above)
Comment (denniswon) [reply packages/accounts/src/light-account/account.ts:42]: @moldy530 actually we use setSigner method here https://github.com/alchemyplatform/aa-sdk/blob/a08ff18059cd504ae5fff2645b09d64acd7ea39d/packages/accounts/src/light-account/actions/transferOwnership.ts#L58

Comment (moldy530) [reply packages/accounts/src/light-account/account.ts]: chatted offline, but I think we should remove the `setSigner` from `transferOwnership` altogether and either:
1) update the docs to reflect that on successful UO, devs need to then call `createLightAccount` to get a new instance connected to the new client
OR
2) update the return type for `transferOwnership` to be something like `{hash: Hash; account: LightAccount}` to make it easier for devs

File: packages/accounts/src/light-account/account.ts [hunk 6]
```diff
@@ -196,13 +193,14 @@ export async function createLightAccount({
       });
     },
     signUserOperationHash: async (uoHash: Hex) => {
-      return owner.signMessage({ raw: uoHash });
+      return signer.signMessage({ raw: uoHash });
     },
     async signMessage({ message }) {
       const version = await getLightAccountVersion(account);
       switch (version) {
+        case "v1.1.0":
```
Comment (denniswon) [inline packages/accounts/src/light-account/account.ts:201]: @moldy530

File: packages/accounts/src/light-account/account.ts [hunk 7]
```diff
@@ -214,8 +212,9 @@ export async function createLightAccount({
     async signTypedData(params) {
       const version = await getLightAccountVersion(account);
       switch (version) {
+        case "v1.1.0":
```
Comment (denniswon) [inline packages/accounts/src/light-account/account.ts:215]: @moldy530

File: packages/accounts/src/light-account/account.ts (hunk 6 above)
Comment (moldy530) [reply packages/accounts/src/light-account/account.ts:201]: handled in the default case below

File: packages/accounts/src/light-account/account.ts (hunk 7 above)
Comment (moldy530) [reply packages/accounts/src/light-account/account.ts:215]: handled in the default case below

Comment (moldy530) [inline packages/accounts/src/light-account/account.ts:215]:
```suggestion
```

Comment (denniswon) [reply packages/accounts/src/light-account/account.ts:215]: ah sorry nvm. didn't see that line.

**Review by moldy530 — DISMISSED** [review]
LGTM, just make sure to run `e2e` tests to make sure nothing broke here

**Comment by denniswon** [comment]
@moldy530 fixed some e2e tests. 

One test failing in e2e is:

```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: {
  "code": -32521,
  "details": "user operation's call reverted: 0xa24a13a6",
  "docsPath": undefined,
  "metaMessages": [
    "URL: https://eth-sepolia.g.alchemy.com/v2/wezhr4S1VfAMhjFXpIb7opdCZcIOPAow",
    "Request body: {"method":"eth_estimateUserOperationGas","params":[{"initCode":"0x00004ec70002a32400f8ae005a26081065620d205fbfb9cf0000000000000000000000007cbfe93f3722c53a8900044d61908e5decac23d60000000000000000000000000000000000000000000000000000000000000000","sender":"0x92496cD1D7afbd516932162509Edea80D94570de","nonce":"0x0","callData":"0xb61d27f600000000000000000000000092496cd1d7afbd516932162509edea80d94570de0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000002a44f1ef2860000000000000000000000000000006641db11037b7a187c45d9d50f2aeee75000000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000224e69e24a8000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000e8f14a838a00505d861c6ef15cdfb05455000000000000000000000000000000000000000000000000000000000000018000000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000121d0f6cca8bef15ac65bf93ae5ae9ec383ea5010bf3e02b44d3c0abd2d2bd0fb0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000c00000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000010000000000000000000000007cbfe93f3722c53a8900044d61908e5decac23d600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","signature":"0xfffffffffffffffffffffffffffffff0000000000000000000000000000000007aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa1c","paymasterAndData":"0x","maxFeePerGas":"0xd999cfe","maxPriorityFeePerGas":"0xa8"},"0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789"]}",
  ],
  "shortMessage": "RPC Request failed.",
  "version": "viem@2.7.8",
  "walk": "Function<walk>",
}
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

 Test Files  1 failed (1)
      Tests  1 failed | 11 passed (12)
   Start at  17:46:56
   Duration  172.46s (transform 287ms, setup 16ms, collect 622ms, tests 171.49s, environment 0ms, prepare 39ms)
   ```
   
   is this rpc endpoint working all well? for sepolia cc: @dancoombs

**Pull request merged**